### DISCUSSION TITLE
i cant use ctx in my code

### DISCUSSION THREAD

**Discussion by mansan1230** [Rapptz/discord.py#7725] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/84826972/159126733-fbc6d214-aa59-47c9-979d-3a6570e59a87.png)
i have follow the https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html?highlight=ctx
but i cant use ctx in my bot 
![image](https://user-images.githubusercontent.com/84826972/159126767-50e8ddf3-36ce-4743-9c3c-d6bcc66f1ec3.png)

**Comment by LeoCx1000** [comment, marked as answer]
Do you have an `on_message` event? If so you need to [process commands in it or use a listener](https://discordpy.readthedocs.io/en/latest/faq.html?highlight=on_message#why-does-on-message-make-my-commands-stop-working)
Are you using v2.0? If so you need to enable [message content intents](https://discordpy.readthedocs.io/en/master/api.html?highlight=message_content#discord.Intents.message_content)

If not, please share more of the code, that's the two things I could think of with the limited info provided.

**Reply by mansan1230** [reply]
import discord 
from discord.ext import commands
import json
intents = discord.Intents.default()
intents.members = True
intents.message_content = True
intents.typing = True


with open('setting.json',encoding='utf8') as jfile:
    jdate = json.load(jfile)

bot = commands.Bot(command_prefix='$', intents=intents)



@bot.event
async def on_ready():
    print(">Bot is online<")

@bot.event
async def on_message(msg):
    channel = bot.get_channel(int(jdate["Test_Channel"]))
    keyword = 'Congratulations to'

    if msg.content.startswith(keyword):
        members = msg.mentions 
        embeds_ = msg.embeds
        str1 = "" 
        display = embeds_[0].to_dict() 
        

        thread_name = str((display['description']))
        start = thread_name.find("[")
        end = thread_name.find("]")
        real_th_name =  (thread_name[start+1:end])
        channel = bot.get_channel(int(jdate["Thread_Channel"]))
        thread = await channel.create_thread(name=real_th_name+" winner",auto_archive_duration=10080)      
        for dc_name in members:
            await thread.send(dc_name.mention)

@bot.command()
async def wl(ctx, msg):
    await ctx.send(msg)


     
bot.run(jdate["TOKEN"])

**Reply by helish88** [reply]
https://discordpy.readthedocs.io/en/latest/faq.html#why-does-on-message-make-my-commands-stop-working
